inc/parse_bonus.h

```diff
@@ -0,0 +1,222 @@
+ * @param ambient	Struct where to save.
+ */
+void	ft_parse_ambient(char *line, t_light *ambient);
+
+/**
+ * @brief Parse camera line.
+ *
+ * Function gets called if line has identifier "C".
+ * Parse the following:
+ * - x,y,z coordinates of the view point.
+ * - 3d normalized orientation vector. In range [-1,1] for each x,y,z axis.
+ * - FOV : Horizontal field of view in degrees in range [0,180].
+ * @param line	Pointer to current line.
+ * @param cam	Struct where to save.
+ */
+void	ft_parse_camera(char *line, t_cam *cam);
+
+/**
+ * @brief Parse light line.
+ *
+ * Function gets called if line has identifier "L".
+ * Parse the following:
+ * - x,y,z coordinates of the light point.
+ * - the light brightness ratio in range [0.0,1.0].
+ * - R,G,B colors in range [0-255].
+ * @param line	Pointer to current line.
+ * @param light	Struct where to save.
+ * @param id	Current light id.
+ */
+void	ft_parse_light(char *line, t_light *light, int id);
+
+// parse_entity2.c
+
+/**
+ * @brief Parse sphere line.
+ *
+ *Function gets called if line has identifier "sp".
+ * Parse the following:
+ * - x,y,z coordinates of the sphere center.
+ * - the sphere diameter.
+ * - the shininess.
+ * - the reflectivity.
+ * - whether it's checkered.
+ * - R,G,B colors in range [0-255].
+ * Also sets type and id.
+ * @param line		Pointer to current line.
+ * @param sphere	Struct where to save.
+ * @param id		Current shape id.
+ */
+void	ft_parse_sphere(char *line, t_hittable *sphere, size_t id);
+
+/**
+ * @brief Parse plane line.
+ *
+ * Function gets called if line has identifier "pl".
+ * Parse the following:
+ * - x,y,z coordinates of a point in the plane.
+ * - 3d normalized normal vector. In range [-1,1] for each x,y,z axis.
+ * - the shininess.
+ * - the reflectivity.
+ * - whether it's checkered.
+ * - R,G,B colors in range [0-255].
+ * Also sets type and id.
+ * @param line	Pointer to current line.
+ * @param plane	Struct where to save.
+ * @param id	Current shape id.
+ */
+void	ft_parse_plane(char *line, t_hittable *plane, size_t id);
+
+/**
+ * @brief Parse cylinder line.
+ *
+ * Function gets called if line has identifier "cy".
+ * Parse the following:
+ * - x,y,z coordinates of the center of the cylinder.
+ * - 3d normalized axis vector of cylinder. In range [-1,1] for each x,y,z axis.
+ * - the cylinder diameter.
+ * - the cylinder height.
+ * - the shininess.
+ * - the reflectivity.
+ * - whether it's checkered.
+ * - R,G,B colors in range [0,255].
+ * Also sets type and id.
+ * @param line		Pointer to current line.
+ * @param cylinder	Struct where to save.
+ * @param id		Current shape id.
+ */
+void	ft_parse_cylinder(char *line, t_hittable *cylinder, size_t id);
+
+/**
+ * @brief Parse cone line.
+ * 
+ * Function gets called if line has identifier "co".
+ * Parse the following:
+ * - x,y,z coordinates of the apex of the cone.
+ * - 3d normalized axis vector of cone. In range [-1,1] for each x,y,z axis.
+ * - the cone radius.
```

Comment:
Yeah of course store it as radius.

It's a legacy question: we have two existing things which want a diameter. Even though that's bad design we still need to adhere to the standard (we solve this issue by just storing it as radius).

Now we add a new thing. Since the old two things want diameter it would be more uniform to ask the user for diameter of the cone. He has no idea that that's actually bad, but he would see 
- cylinder->diameter; 
- sphere->diameter; 
- cone->radius.

Reply:
So you wanna put diameter for the cone in the .rt file, but store and process it behind the scenes as radius? What about your display in the menu? There it should be again diameter?

Reply:
Yes, as the same thing happens for sphere and cylinder.

Display should be in diameter again. I'd argue we don't need an extra variable for  it, since the calc *2 only happens once per menu page and not per pixel